libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -1483,6 +1485,28 @@ - (void)switchToNewUser {
     [self switchToUser:nil];
 }
 
+- (void)switchToNewUserWithCompletion:(void (^)(NSError *error, SFUserAccount * currentAccount))completion {
+    if (!self.currentUser) {
```

Comment:
Not sure I understand this. Why can't we switch to a new user without `currentUser` being set? This is basically the deferred authentication scenario, where there's a playground mode, followed by an API call to this API. At this point, `self.currentUser` will be `nil`.

Reply:
@bhariharan  The check is to ensure that there is a logged in user at the least i.e. the switch is occurring from the context of  a logged in user.  Why would this call be made otherwise vs. using loginWithCompletion.

Reply:
Well, I was thinking they'd be the same. Basically deprecate `loginWithCompletion` in this class and rename it to `switchToNewUser`. That makes the APIs sound the same - `switchToUser` and `switchToNewUser`. Also, `switchToUser` would fall back on `switchToNewUser` if no user is passed in. Makes sense?

Reply:
@bhariharan I don't think we should. To start with login api has the advantage of being concise in what the api means and does, as opposed to "switch to a new User". Secondly the login api **does not set the current user** or technically speaking  "switch to the user".  A developer could set the current user to any of the logged in users. Third reason is that the login api and logout together symmetry.

We could rename the api to loginAndSwitchUser instead? A more radical approach would be to consider the addition of a  user to the UserAccountManager as a login? So then the developer would always add a user  account ->  set the current user or switch user. Perhaps we should disallow the setting the currentUser and resort to switchUser at all times.  We should chat about it offline.

Reply:
@trooper2013 The whole point of this API is to avoid having the app set `currentUser`, or expecting the app developer to handle that case. Basically, the end result of `loginWithCompletion` and `switchToNewUser` should be the same - we should set `currentUser` to that newly logged in user automatically, and then pass along control to the app by calling their completion blocks. I can't think of many use cases where an app would want to login with a new user and then not switch to that user. Login should automatically land the user on that newly logged in user's state.

I'm okay with keeping both `loginWithCompletion` and `switchToNewUser` the way they are - let's make sure that all the methods make setting of `currentUser` internal to the SDK.